src/api/galleryItem.controller.ts

```diff
@@ -21,11 +21,21 @@ router.post("/", (req, res) => {
 });
 
 router.get("/", async (req, res) => {
-  try {
-    const galleryItems = await repository.getAll();
-    res.status(200).json(galleryItems);
-  } catch (error: any) {
-    res.status(500).json({ message: error.message });
+  const limit = req.query.limit
+    ? parseInt(req.query.limit as string)
+    : undefined;
+  if (limit && (isNaN(limit) || limit < 0)) {
+    res.status(400).json({
+      message:
+        "Invalid 'limit' query parameter. Must be a non-negative integer.",
+    });
+  } else {
```

Comment:
As an improvement, instead of using if else block here, you could benefit by returning early if the validation has failed. For example

```suggestion
  if (limit && (isNaN(limit) || limit < 0)) {
    return res.status(400).json({
      message:
        "Invalid 'limit' query parameter. Must be a non-negative integer.",
    });
  }
  // continue the flow without the else block
```

Reply:
Thank youu, express expects the return to be void (I think) so i returned after the response 🫡